packages/api/src/folder/models.ts

```diff
@@ -0,0 +1,107 @@
+export interface Folder {
+  folderId: string;
+  archiveNumber: string;
+  archiveId: string;
+  displayName: string;
+  downloadName?: string;
+  downloadNameOk?: boolean;
+  displayDate?: string;
+  displayEndDate?: string;
+  timeZoneId?: bigint;
+  note?: string;
+  description?: string;
+  special?: string;
+  sort?: string;
+  locationId?: bigint;
+  view?: string;
+  viewProperty?: string;
+  thumbArchiveNumber?: string;
+  imageRatio?: number;
+  thumbStatus?: string;
+  thumbUrl200?: string;
+  thumbUrl500?: string;
+  thumbUrl1000?: string;
+  thumbUrl2000?: string;
+  status: FolderStatus;
+  type: FolderType;
+  publicAt: string;
+  createdAt: string;
+  updatedAt?: string;
+  thumbnail256?: string;
+  thumbnail256CloudPath?: string;
+}
+
+export interface FolderRow {
```

Comment:
What is the difference between this and `Folder`?

Reply:
There is no difference, I saw that we use the same approach for the ArchiveRecord and ArchiveRecordRow and used the same approach. I thought we use one as a datamodel and one for the result of the query.

Reply:
The `record` table has some columns where `bigint`s are used to represent actual numbers, rather than IDs. Because `bigint`s exceed the maximum size of 32-bit integers, `tinypg` maps them to strings, which we want for IDs, but not for actual numbers. `ArchiveRecordRow` exists to represent what we get out of the database, which we then convert to the actual data model of `ArchiveRecord`. In the case where we can get data matching the data model directly from the database, I don't think we need a separate interface for the database output (we can always add one if the situation changes).